ONSdigital/eq-publisher-v3
# EAR-1394

What is the context of this PR?

Abstracted out Place Holder Array Builder

- Abstracted out the Place Holder Array Builder so that it now returns the correct placeholder object depending on the type that's being passed in.
- Values were being passed through multiple levels which seemed a little redundant. Abstracted out and hopefully simplified it.
- Renamed some variables so that they make more sense in the context they're being used.
- Removed the `PlaceHolderText` because as far as i could see, it was always `output`.
- Removed `FILTER_MAP` stuff, to a `pipeAnswerCheck`. I think simplifies the code on top of that makes it a bit easier to read.

How to review

* Run v3 of the projects. In `eq-author` pass `hub` to `REACT_APP_FEATURE_FLAGS`.
* Pip through data to check that its working.
* Create a date, don’t enter the date and check that it defaults to the meta data value.
* Enter in a date to check that it pipes in the date you entered.
* Check that piped in author will default to the meta data if there is no author entered.



What to do after everything is green
    *   Bring this branch up-to-date with Origin/Master
    *   If there are a lot of commits or some are not helpful to read, squash them down
    *   Click the Merge button on this pull request
    *   Does this change mean the Capability examples questionnaire should be updated?
    *   Move the Jira ticket for this task into the next stage of the process
